internal/engine/rtengine/cache.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rtengine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/engine/ingestcache"
+	"github.com/stacklok/minder/internal/ruletypes"
+	provinfv1 "github.com/stacklok/minder/pkg/providers/v1"
+)
+
+// Cache contains a set of RuleTypeEngine instances
+type Cache interface {
+	GetRuleEngine(ruleTypeID uuid.UUID) (*RuleTypeEngine, error)
+}
+
+type ruleEngineCache struct {
+	engines map[uuid.UUID]*RuleTypeEngine
```

Comment:
should this be a [syncmap](https://pkg.go.dev/golang.org/x/sync/syncmap) instead so this could be updated/read in parallel?

Reply:
Discussed this with @JAORMX offline: the current implementation populates all the rule type instances up front. As a result, we do not need to worry about concurrent read/write scenarios.